primeqa/mrc/README.md

```diff
@@ -249,13 +249,61 @@ Additionally, to specify a MRQA subset e.g. `SQuAD`, `NaturalQuestionsShort`, `T
        --dataset_filter_column_name subset
 ```
 
-Cross domain experiments can be run by running train and eval as separate processes. The
+Cross domain experiments can be run by running train and eval as separate processes.
 
  
 ### Custom Data
 
 Users can also train (fine-tune) and evaluate the MRC model on custom data by providing their own train_file and eval_file. Instructions for getting started are available [here](../../examples/custom_mrc/README.md).
 
+
+### Training with Multiple Datasets
+
+PrimeQA supports the training of MRC model with combination of multiple datasets, which are specified in "--train_fof" argument. Three formats, csv, jsonl and json, can be used to define "--train_file".
+
+In csv format, the columns of each line, separated by space, are for:
+ - HuggingFace dataset name, or path of local data file, or path of dataset processing script (in python);
+ - Dataset config name or data file format;
+ - Sampling rate within range 0.0 to 1.0, e.g. 0.5 means 50% of the examples are randomly selected and used in MRC training;
+ - Preprocessor name.
+
+If column 2 to 4 are not given, default values from input arguments will be used, i.e.:
+ - Value of "--dataset_config_name" for HuggingFace dataset and processing script, or "--data_file_format" for local data file;
+ - 1.0 for sampling rate;
+ - Value of "--preprocessor" for preprocessor name.
+
+If in jsonl format, each line is a dictionary consisting of
+```
+{'dataset': dataset_name_or_path_of_data_file_or_processing_script,
+ 'config': dataset_config_or_data_file_format,
+ 'sampling_rate': sampling_rate,
+ 'preprocessor': preprocessor_name}
+```
+
+Fields 'config', 'sampling_rate', and 'preprocessor' are optional.
+
+If in json format, a list of dictionaries same to that in jsonl is expected.
+
+The following is an example of "--train_fof" in csv format which includes two HuggingFace datatsets: TyDiQA and SQuAD.
+```
+tydiqa primary_task 0.1 primeqa.mrc.processors.preprocessors.tydiqa.TyDiQAPreprocessor
+squad  plain_text   0.1 primeqa.mrc.processors.preprocessors.squad.SQUADPreprocessor
+```
+
+To evaluate the checkpoint models during training, the validation dataset needs be specified in "--eval_fof" which format is same to "--train_fof". Multiple datatsets can be included into "--eval_fof", but only the first one is used for evaluation.
```

Comment:
"Multiple datatsets can be included into "--eval_fof", but only the first one is used for evaluation.":

1. This is only true for the base trainer, right? We should specifically say that here.
2. For the base trainer, shouldn't we at least merge all the eval datasets provided by the user and treat the merged dataset as a single large eval dataset?

Reply:
I don't understand this. If only one is used for evaluation why can you include many?

Reply:
@ma-sultan @seirasto Yes, we can evaluate multi validation sets in run_mrc, but this requires that
1. postprocessor and eval_metrics have to be included in eval_fof;
2. multi trainers need be created, since HuggingFace trainer class only allows one "post_process_function" and one "compute_metrics". 
So I'd like to use a separate PR to address this if necessary. Now users can still run eval jobs for different validation sets separately.

@ma-sultan "This is only true for the base trainer, right? We should specifically mention that here. "  --- changed the README to reflect this.